app/controllers/users_controller.rb

```diff
@@ -60,9 +60,9 @@ def preferences
 
    # /users/1/switch
   def switch
-    authorize User
+    @user = User.find(params[:id])
+    authorize @user
```

Comment:
Since you are in this switch code. It would be good to just remove the `get` part of this action. There probably is a corresponding view. We don't use it anymore.